src/components/context/ErrorContext.tsx

```diff
@@ -0,0 +1,39 @@
+import React, {
+  Dispatch,
+  SetStateAction,
+  createContext,
+  useContext,
+  useState,
+} from 'react';
+import { TodoError } from '../../types/enums';
+
+interface ErrorContextType {
+  errorMessage: TodoError | null;
+  setErrorMessage: Dispatch<SetStateAction<TodoError | null>>;
+}
+
+const ErrorContext = createContext<ErrorContextType | null>(null);
+
+interface Props {
+  children: React.ReactNode;
+}
+
+export const ErrorProvider: React.FC<Props> = ({ children }) => {
+  const [errorMessage, setErrorMessage] = useState<TodoError | null>(null);
```

Comment:
Better not to use `null` but to create default option with empty message inside `TodoError`